cadc-download-manager-server/src/main/java/ca/nrc/cadc/dlm/server/ShellScriptServlet.java

```diff
@@ -0,0 +1,186 @@
+    private static final long serialVersionUID = 202303151015L;
+    public static final String SCRIPT_TARGET = "/shellScript";
+
+    // Two (2) weeks in minutes
+    private static final int MINUTES_TO_EXPIRY = 60 * 24 * 7 * 2;
+
+    private static final DateFormat DATE_FORMAT = DateUtil.getDateFormat("yyyyMMddHHmmss", DateUtil.UTC);
+
+    /**
+     * Handle POSTed download request from an external page.
+     *
+     * @param request  The HTTP Request
+     * @param response The HTTP Response
+     * @throws IOException if stream processing fails
+     */
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+
+        final String filename = String.format("cadc-download-%s.sh", DATE_FORMAT.format(new Date()));
+        response.setContentType("text/x-shellscript");
+        response.setHeader("content-disposition", "attachment;filename=\"" + filename + "\"");
+
+        final DownloadRequest downloadReq = (DownloadRequest) request.getAttribute("downloadRequest");
+        downloadReq.runID = (String) request.getAttribute("runid");
+
+        final Iterator<DownloadDescriptor> downloadDescriptorIterator = DownloadUtil.iterateURLs(downloadReq);
+        final Subject currentSubject = AuthenticationUtil.getSubject(request, false);
+        final ScriptGenerator scriptGenerator;
+
+        try {
+            if (CredUtil.checkCredentials(currentSubject)) {
+                final String authToken = getToken(currentSubject);
+                final Calendar calendar = Calendar.getInstance(DateUtil.UTC);
+
+                // Expiry is current time plus the set expiry minutes.
+                calendar.add(Calendar.MINUTE, ShellScriptServlet.MINUTES_TO_EXPIRY);
+                scriptGenerator = new ScriptGenerator(downloadDescriptorIterator, authToken, calendar.getTime());
+            } else {
+                scriptGenerator = new ScriptGenerator(downloadDescriptorIterator);
+            }
+
+            final Writer writer = response.getWriter();
+            scriptGenerator.generate(writer);
+            writer.flush();
+        } catch (CertificateExpiredException | CertificateNotYetValidException certificateException) {
+            throw new AccessControlException(certificateException.getMessage());
+        } catch (IOException ioException) {
+            throw ioException;
+        } catch (Exception exception) {
+            throw new ServletException(exception.getMessage(), exception);
+        }
+    }
+
+    private String getToken(final Subject subject) throws Exception {
+        final LocalAuthority localAuthority = new LocalAuthority();
+        final URI oAuthServiceURI = localAuthority.getServiceURI(Standards.SECURITY_METHOD_OAUTH.toString());
+        final RegistryClient registryClient = new RegistryClient();
+        final URL baseOAuthServiceURL = registryClient.getServiceURL(oAuthServiceURI, Standards.SECURITY_METHOD_OAUTH,
```

Comment:
Given the presence of the OAUTH endpoint in LocalAuthority.properties and in the /ac capabilities, and given that the accessURL points to the /authorize endpoint, this is currently the right way to find the URL.  However, we are wondering if that OAUTH endpoint should exist at all (it is not an API) and whether it could be replaced by the OPENID endpoint.  cc @pdowler